transmart-rest-api/src/main/groovy/org/transmartproject/rest/serialization/tabular/TabularResultSPSSSerializer.groovy

```diff
@@ -121,25 +122,54 @@ class TabularResultSPSSSerializer implements TabularResultSerializer {
         Iterator<DataRow> rows = tabularResult.rows
         while (rows.hasNext()) {
             DataRow row = rows.next()
-            List<Object> valuesRow = columns.stream().map({ DataColumn column -> row[column] }).collect(Collectors.toList())
-            csvWriter.writeNext(formatRowValues(valuesRow))
+            String[] formattedValuesRow = getFormattedValuesRow(columns, row)
+            csvWriter.writeNext(formattedValuesRow)
         }
         csvWriter.flush()
     }
 
-    private String[] formatRowValues(List<Object> valuesRow) {
-        valuesRow.stream().map({value ->
-            if (value == null) return ''
-            if (value instanceof Date) {
-                synchronized (DATE_FORMAT) {
-                    DATE_FORMAT.format(value)
-                }
-            } else {
-                value.toString()
-            }
+    private String[] getFormattedValuesRow(ImmutableList<DataColumn> columns, DataRow row) {
+        columns.stream().map({ DataColumn column ->
+            getFormattedValue(column, row)
         }).collect(Collectors.toList()).toArray(new String[0])
     }
 
+    private String getFormattedValue(DataColumn column, DataRow row) {
+        VariableMetadata metadata = getColumnMetadata(column)
+        def value = row[column]
+        return formatRowValue(value, metadata)
+    }
+
+    private static VariableMetadata getColumnMetadata(DataColumn column) {
+        if (column instanceof MetadataAwareDataColumn && ((MetadataAwareDataColumn)column).metadata) {
+            return ((MetadataAwareDataColumn)column).metadata
+        } else {
+            return null
+        }
+    }
+
+    private String formatRowValue(Object value, VariableMetadata metadata) {
+        if (value == null)
+            return ''
+        else if (value instanceof Date) {
+            return formatDateRowValue((Date)value, metadata)
+        } else {
+            return value.toString()
+        }
+    }
+
+    private String formatDateRowValue(Date value, VariableMetadata metadata) {
+        if (metadata.type == VariableDataType.DATETIME) {
+            synchronized (DATETIME_FORMAT) {
+                return DATETIME_FORMAT.format(value)
+            }
+        } else {
```

Comment:
maybe it makes more sense to return datetime format by default?

Reply:
good point, I changed it